app/src/main/java/io/left/meshim/controllers/RightMeshController.java

```diff
@@ -63,9 +68,14 @@
     private IActivity callback = null;
     //reference to service
     private MeshIMService meshIMService;
-    // keeps track of all the undeliveredMessages.
+    // keeps track of all the undeliveredPackages.
     private HashMap<Integer, Integer> unDeliveredMessageIds = new HashMap<Integer, Integer>();
+    private ConcurrentMap<Integer,MeshId> undeliveredPeerUpdateMessage = new ConcurrentHashMap<>();
```

Comment:
Plural name to indicate this holds many of these.

Reply:
been meaning to change that since yesterday.. :expressionless: